projects/packages/forms/src/blocks/contact-form/components/jetpack-integrations-modal/integration-card/integration-card-header.tsx

```diff
@@ -67,11 +69,11 @@ const IntegrationCardHeader = ( {
 		setupBadge,
 	} = cardData;
 	const showPluginAction = type === 'plugin' && ( ! isInstalled || ! isActive );
-	const showConnectedBadge = isActive && isConnected;
+	const showConnectedBadge = isConnected || ( isActive && ! needsConnection );
```

Comment:
This will show Enabled badge for a plugin is it is active and does not need a connection.